lib/screens/v2/candidate_generator/on_bus/departures.ex

```diff
@@ -80,6 +91,29 @@ defmodule Screens.V2.CandidateGenerator.Widgets.OnBus.Departures do
     Enum.filter(departures, &(&1.prediction.route.id != route_id))
   end
 
+  defp filter_duplicate_routes(departures) do
+    unique_departures =
+      Enum.uniq_by(departures, fn dep ->
+        {dep.prediction.route.line.id, dep.prediction.trip.direction_id,
+         Trip.representative_headsign(dep.prediction.trip)}
+      end)
+
+    # If there are fewer than 3 unique connecting departures, then return at least 3 based on soonest departure time.
+    case length(unique_departures) do
+      count when count >= 3 ->
```

Comment:
A `case` with only one condition and a fallback might be more naturally expressed as an `if`:

```elixir
if length(unique_departures) >= 3 do
  unique_departures
else
  # ...
end
```